src/services/reportCache.js

```diff
@@ -164,6 +166,61 @@ const cacheResources = async (objectiveId, activityReportObjectiveId, resources
   ]);
 };
 
+const cacheCourses = async (objectiveId, activityReportObjectiveId, courses = []) => {
+  const courseIds = courses.map((course) => course.courseId);
+  const courseSet = new Set(courseIds);
+  const originalAroCourses = await ActivityReportObjectiveCourse.findAll({
+    where: { activityReportObjectiveId },
+    raw: true,
+  });
+  const originalCourseIds = originalAroCourses.map((course) => course.courseId)
+    || [];
+  const removedCourseIds = originalCourseIds.filter((id) => !courseSet.has(id));
+  const currentCourseIds = new Set(originalCourseIds.filter((id) => courseSet.has(id)));
+  const newCourseIds = courseIds.filter((id) => !currentCourseIds.has(id));
+
+  return Promise.all([
+    ...newCourseIds.map(async (courseId) => ActivityReportObjectiveCourse.create({
```

Comment:
Should we be running individual hooks on creation also?

Reply:
Leaving it, since this is identical to how we do it on topics, resources, etc